src/engine/QueryExecutionTree.cpp

```diff
@@ -92,19 +92,31 @@ void QueryExecutionTree::setVariableColumns(
 // _____________________________________________________________________________
 template <QueryExecutionTree::ExportSubFormat format>
 ad_utility::stream_generator::stream_generator
-QueryExecutionTree::generateResults(const SelectedVarsOrAsterisk & selectVarsOrAsterisk,
+QueryExecutionTree::generateResults(const SelectedVarsOrAsterisk & selectedVarsOrAsterisk,
                                     size_t limit, size_t offset) const {
   // They may trigger computation (but does not have to).
   shared_ptr<const ResultTable> resultTable = getResult();
   LOG(DEBUG) << "Resolving strings for finished binary result...\n";
   vector<std::optional<pair<size_t, ResultTable::ResultType>>> validIndices;
-  if(selectVarsOrAsterisk.isAsterisk()) {
-    for(const auto& elem : getVariableColumns())
-      validIndices.emplace_back(pair<size_t, ResultTable::ResultType>(
-          elem.second, resultTable->getResultType(elem.second)));
+  if(selectedVarsOrAsterisk.isAsterisk()) {
+    list<string> orderedVariables = selectedVarsOrAsterisk.retrieveOrder();
+    auto allVars = getVariableColumns();
+    for (const auto& var : orderedVariables) {
+      auto it = allVars.find(var);
+      if (it != allVars.end()) {
+        validIndices.emplace_back(pair<size_t, ResultTable::ResultType>(
+            it->second, resultTable->getResultType(it->second)));
+        allVars.erase(it);
+      } else {
+        validIndices.emplace_back(std::nullopt);
+      }
+    }
+    for(const auto& var : allVars){
+      LOG(DEBUG) << "Variable " << var.first << " was not parsed!! \n";
```

Comment:
`LOG(WARN) << "The variable \"" << var.first << "\" was found in the execution tree, but not in the original query. This is likely a bug\n";`